ovos_bus_client/util/scheduler.py

```diff
@@ -328,19 +331,23 @@ def shutdown(self):
         """
         Stop the running thread.
         """
-        self._stopping.set()
-        self._running.clear()
-        # Remove listeners
-        self.bus.remove_all_listeners('mycroft.scheduler.schedule_event')
-        self.bus.remove_all_listeners('mycroft.scheduler.remove_event')
-        self.bus.remove_all_listeners('mycroft.scheduler.update_event')
-        # Wait for thread to finish
-        self.join()
-        # Prune event list in preparation for saving
-        self.clear_repeating()
-        self.clear_empty()
-        # Store all pending scheduled events
-        self.store()
+        try:
+            self._stopping.set()
+            # Remove listeners
+            self.bus.remove_all_listeners('mycroft.scheduler.schedule_event')
+            self.bus.remove_all_listeners('mycroft.scheduler.remove_event')
+            self.bus.remove_all_listeners('mycroft.scheduler.update_event')
+            # Wait for thread to finish
+            self.join()
+            # Prune event list in preparation for saving
+            self.clear_repeating()
+            self.clear_empty()
+            # Store all pending scheduled events
+            self.store()
+        except Exception as e:
+            LOG.exception(e)
```

Comment:
Should the process attempt to quit itself ungracefully in this case?

Reply:
Hmm, I was thinking mostly that the `self._running` event should be cleared in any case since I believe the scheduler should be considered "not running"

Reply:
FWIW, the only errors I would expect here are:
- `self.join` fails because the thread wasn't started
- `self.store` fails because of FS permissions